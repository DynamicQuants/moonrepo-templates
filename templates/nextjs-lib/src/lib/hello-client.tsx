import styles from './hello-client.module.css';

export function HelloClient() {
  return (
    <div className={styles['container']}>
      <h1>Hello from Client!</h1>
    </div>
  );
}

export default HelloClient;
