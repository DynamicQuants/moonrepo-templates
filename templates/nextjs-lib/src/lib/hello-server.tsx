// React server components are async so you make database or API calls.
export async function HelloServer() {
  return (
    <div>
      <h1>Hello from Server!</h1>
    </div>
  );
}
