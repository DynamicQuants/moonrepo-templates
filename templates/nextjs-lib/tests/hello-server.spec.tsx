import { cleanup, render, screen } from '@testing-library/react';

import { HelloServer } from '../src/lib/hello-server';

describe('HelloServer', () => {
  beforeEach(async () => {
    render(await HelloServer());
  });

  afterEach(() => {
    cleanup();
  });

  it('should match the snapshot', async () => {
    const { container } = render(await HelloServer());
    expect(container).toMatchSnapshot();
  });

  it('should have h1 with correct text', () => {
    const heading = screen.getByRole('heading', { name: 'Hello from Server!' });
    expect(heading).toBeInTheDocument();
  });
});
