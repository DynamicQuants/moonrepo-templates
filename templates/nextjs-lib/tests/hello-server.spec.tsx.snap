// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloServer should match the snapshot 1`] = `
<div>
  <div>
    <h1>
      Hello from Server!
    </h1>
  </div>
</div>
`;
